src/Elasticsearch.Net/ConnectionPool/SniffingSortedStickyConnectionPool.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Elasticsearch.Net
+{
+	public class SniffingSortedStickyConnectionPool : IConnectionPool
+	{
+		private readonly ReaderWriterLockSlim _readerWriter = new ReaderWriterLockSlim();
+
+		private readonly Func<IEnumerable<Node>, IEnumerable<Node>> _sortMethod;
+
+		protected IDateTimeProvider DateTimeProvider { get; }
+
+		protected List<Node> InternalNodes { get; set; }
+
+		/// <inheritdoc/>
+		public bool UsingSsl { get; }
+
+		/// <inheritdoc/>
+		public bool SniffedOnStartup { get; set; }
+
+		/// <inheritdoc/>
+		public IReadOnlyCollection<Node> Nodes => this.InternalNodes;
+
+		/// <inheritdoc/>
+		public int MaxRetries => this.InternalNodes.Count - 1;
+
+		/// <inheritdoc/>
+		public bool SupportsReseeding => true;
+
+		/// <inheritdoc/>
+		public bool SupportsPinging => true;
+
+		/// <inheritdoc/>
+		public DateTime LastUpdate { get; protected set; }
+
+		public SniffingSortedStickyConnectionPool(IEnumerable<Uri> uris, Func<IEnumerable<Node>, IEnumerable<Node>> sortMethod, IDateTimeProvider dateTimeProvider = null)
+			: this(uris.Select(uri => new Node(uri)), sortMethod, dateTimeProvider)
+		{
+		}
+
+		public SniffingSortedStickyConnectionPool(IEnumerable<Node> nodes, Func<IEnumerable<Node>, IEnumerable<Node>> sortMethod, IDateTimeProvider dateTimeProvider = null)
+		{
+			nodes.ThrowIfEmpty(nameof(nodes));
+
+			this.DateTimeProvider = dateTimeProvider ?? Elasticsearch.Net.DateTimeProvider.Default;
+
+			var nn = nodes.ToList();
+			var uris = nn.Select(n => n.Uri).ToList();
+			if (uris.Select(u => u.Scheme).Distinct().Count() > 1)
+				throw new ArgumentException("Trying to instantiate a connection pool with mixed URI Schemes");
+
+			this.UsingSsl = uris.Any(uri => uri.Scheme == "https");
+
+			this.InternalNodes = nn
+				.DistinctBy(n => n.Uri)
+				.ToList();
+			this.LastUpdate = this.DateTimeProvider.Now();
+
+			this._sortMethod = sortMethod;
+		}
+
+		protected int GlobalCursor = -1;
+
+		/// <summary>
+		/// Creates a view of all the live nodes preserving their order,
+		/// if there are no live nodes yields a different dead node to try once
+		/// </summary>
+		public IEnumerable<Node> CreateView(Action<AuditEvent, Node> audit = null)
```

Comment:
Is it possible to extend this class from sniffing connection pool (and in turn static connection pool) and move the common bits of this routine to protected methods?